warehouse/accounts/services.py

```diff
@@ -172,15 +172,20 @@ def check_password(self, userid, password, *, tags=None):
         return False
 
     def create_user(
-        self, username, name, password, is_active=False, is_superuser=False
+        self,
+        username,
+        name,
+        password,
+        is_active=False,
     ):
 
         user = User(
             username=username,
             name=name,
             password=self.hasher.hash(password),
             is_active=is_active,
-            is_superuser=is_superuser,
+            is_superuser=False,
```

Comment:
After removing the defaults from the interface, I noticed that the default isn't respected with db.Model. This is possibly a bug. This is what you will see if you don't have these lines

```
self = <sqlalchemy.dialects.postgresql.psycopg2.PGDialect_psycopg2 object at 0x7f5f8020e668>, cursor = <cursor object at 0x7f5f7f5ec618; closed: -1>
statement = 'INSERT INTO users (username, name, password, is_active, disabled_for) VALUES (%(username)s, %(name)s, %(password)s, %(is_active)s, %(disabled_for)s) RETURNING users.id'
parameters = {'disabled_for': None, 'is_active': False, 'name': 'test_name', 'password': '$argon2i$v=19$m=1024,t=6,p=6$dG5NSWltzfn//3/vnbP23g$+7Alw6BCiPsC38K1JMkDmw', ...}
context = <sqlalchemy.dialects.postgresql.psycopg2.PGExecutionContext_psycopg2 object at 0x7f5f7ea9bd30>

    def do_execute(self, cursor, statement, parameters, context=None):
>       cursor.execute(statement, parameters)
E       sqlalchemy.exc.IntegrityError: (psycopg2.IntegrityError) null value in column "is_superuser" violates not-null constraint
E       DETAIL:  Failing row contains ($argon2i$v=19$m=1024,t=6,p=6$dG5NSWltzfn//3/vnbP23g$+7Alw6BCiPsC..., 2019-01-09 17:31:33.509108, null, test_user, test_name, f, 2019-01-09 17:31:33.509108, d, 2019-01-09 17:31:33.509108, b4bc1a89-aaf5-47a9-8b25-84a4f6c0d734, null, f).
E        [SQL: 'INSERT INTO users (username, name, password, is_active, disabled_for) VALUES (%(username)s, %(name)s, %(password)s, %(is_active)s, %(disabled_for)s) RETURNING users.id'] [parameters: {'username': 'test_user', 'name': 'test_name', 'password': '$argon2i$v=19$m=1024,t=6,p=6$dG5NSWltzfn//3/vnbP23g$+7Alw6BCiPsC38K1JMkDmw', 'is_active': False, 'disabled_for': None}] (Background on this error at: http://sqlalche.me/e/gkpj)

../lib/python3.6/site-packages/sqlalchemy/engine/default.py:509: IntegrityError
```

Reply:
The issue is that `compare_server_default` is off by default, so an auto-migration for this change wouldn't actually add the defaults. I fixed this in #5275.

Reply:
Thanks @di I will rebase on my end and correct :)